api/envoy/extensions/filters/network/sni_dynamic_forward_proxy/v3alpha/sni_dynamic_forward_proxy.proto

```diff
@@ -28,9 +28,4 @@ message FilterConfig {
   // <envoy_api_field_extensions.clusters.dynamic_forward_proxy.v3.ClusterConfig.dns_cache_config>`.
   common.dynamic_forward_proxy.v3.DnsCacheConfig dns_cache_config = 1
       [(validate.rules).message = {required: true}];
-
-  oneof port_specifier {
-    // The port number to connect to the upstream.
-    uint32 port_value = 2 [(validate.rules).uint32 = {lte: 65535 gt: 0}];
```

Comment:
Drive by: I could imagine we can either use original port from downstream, or the port in the SRV record of dns response. 
Now that we have `oneof port_specifier` here, what you think if we another field called `use_downstream_port` within `port_specifier`?

Reply:
Yeah that could be good, however I couldn't see how to read the filter config in `cluster.cc`. To keep that `port_value` option I think the config would be needed to decide whether to suffix `:<port>` to the host or not.

As for the using the SRV record, I don't know enough about how the DNS cache works, but would it be enough to just not append `:<port>` and use `default_port=0` in `loadDnsCacheEntry`? Or would more work be needed to support SRV records?